### DISCUSSION TITLE
When it displays Chinese, garbled characters occur

### DISCUSSION THREAD

**Discussion by jack9603301** [kovidgoyal/kitty#7061] [Q&A] [answered]
**Describe the bug**
I am located in China, and when using the Linux operating system Kitty to view directories, it always inexplicably starts displaying garbled characters in Chinese at certain times. Is this a font output error? How should I solve it? Please fix this issue as soon as possible

Equal width font used: Source Code Pro

**To Reproduce**
N/A

**Screenshots**
![image](https://github.com/kovidgoyal/kitty/assets/9047180/8c0ec410-a22d-40f9-b6c5-50a5d1f3cb75)

**Environment details**

branch: master
Operator System: Linux From Scratch 12.0

**Comment by kovidgoyal** [comment]
You need to install a font on your system that has glyphs covering
whatever those characters are.

**Comment by jack9603301** [comment]
Is there a recommended font?

**Comment by kovidgoyal** [comment]
Use whatever font you like, remember this is a font you simply need to have installed not the font you configure kitty to use.

**Comment by jack9603301** [comment]
I installed Google Fonts and Source Code Pro
What else do I need?

**Comment by jack9603301** [comment]
![image](https://github.com/kovidgoyal/kitty/assets/9047180/0dd01972-bd54-4aa5-9f06-7497fa26ea99)

**Comment by jack9603301** [comment]
Besides it, I usually rarely use other fonts. Can you give me a direction?

**Reply by kovidgoyal** [reply, marked as answer]
I have no idea, I dont use chinese text. Google for what fonts have chinese symbols and install some. Then you can run kitty with --debug-font-fallback and see what fonts it is using for what symbols.

**Reply by jack9603301** [reply]
![image](https://github.com/kovidgoyal/kitty/assets/9047180/486612a7-4b1c-4752-8e5b-fd7cafa57ce7)
Why isn't it all like this below

**Reply by jack9603301** [reply]
@kovidgoyal I'm trying to use the symbol_map configuration command to override Chinese encoded glyphs using Source Han Serif. It's not an equally wide font, but it actually works. Everything is fine now

**Comment by npc-z** [comment]
I have installed and used the `JetBrainsMono Nerd Font`, not all but some parts of chinese characters have problem to display, when i zoom in or zoom out once whith keyboard `ctrl` + `shift` + `+`  (or `-`), it will to be normal for all, and i have tried different font size.  by the way, i also use the `foot` with the same font, it works well.

```sh
~/.config/kitty on  master! ⌚ 9:18:56
$ neofetch
          ▗▄▄▄       ▗▄▄▄▄    ▄▄▄▖            npc@ser7-nixos 
          ▜███▙       ▜███▙  ▟███▛            -------------- 
           ▜███▙       ▜███▙▟███▛             OS: NixOS 24.05.20240207.f8e2ebd (Uakari) x86_64 
            ▜███▙       ▜██████▛              Host: AZW SER 
     ▟█████████████████▙ ▜████▛     ▟▙        Kernel: 6.7.4 
    ▟███████████████████▙ ▜███▙    ▟██▙       Uptime: 23 hours, 26 mins 
           ▄▄▄▄▖           ▜███▙  ▟███▛       Packages: 1043 (nix-system), 606 (nix-user) 
          ▟███▛             ▜██▛ ▟███▛        Shell: zsh 5.9 
         ▟███▛               ▜▛ ▟███▛         Resolution: 1920x1080 
▟███████████▛                  ▟██████████▙   DE: Hyprland (Wayland) 
▜██████████▛                  ▟███████████▛   WM: .Hyprland-wrapp 
      ▟███▛ ▟▙               ▟███▛            Terminal: kitty 
     ▟███▛ ▟██▙             ▟███▛             Terminal Font: JetBrainsMono Nerd Font 18.0 
    ▟███▛  ▜███▙           ▝▀▀▀▀              CPU: AMD Ryzen 7 7840HS w/ Radeon 780M Graphics (16) @ 5.137GHz 
    ▜██▛    ▜███▙ ▜██████████████████▛        GPU: AMD ATI Phoenix1 
     ▜▛     ▟████▙ ▜████████████████▛         Memory: 2596MiB / 11676MiB 
           ▟██████▙       ▜███▙
          ▟███▛▜███▙       ▜███▙                                      
         ▟███▛  ▜███▙       ▜███▙                                     
         ▝▀▀▀    ▀▀▀▀▘       ▀▀▀▘


~/.config/kitty on  master! ⌚ 9:19:00
$ kitty --version
kitty 0.32.1 created by Kovid Goyal
```

**Comment by SharzyL** [comment]
I experienced the same thing. Part of Chinese (and Japanese) characters were rendered as boxes, and somehow went normal when I zoomed in. It looks like a rendering issue instead of a font selection issue.

Here is a video showing how it looks like:

https://github.com/kovidgoyal/kitty/assets/46294732/91248ccb-e9ff-46e7-bf99-cda832523331

The garbled characters do not always occur. They disappear and reoccur seemly randomly.

**Reply by I-Want-ToBelieve** [reply]
@kovidgoyal Hi，This video looks a lot like a rendering issue, what do you think?
